### ISSUE TITLE
RegressionForest doesn't match fit-time shape of vector Y when predicting

### ISSUE THREAD

**Issue by kbattocchi** [microsoft/EconML#416]
If Y is a vector when fitting a RegressionForest, the return value from predict is an n×1 array instead.  For example:

```python
from econml.grf import RegressionForest
X=np.array([[0,1],[2,3],[4,5]])
y=np.array([1,2,3])
assert RegressionForest().fit(X,y).predict(X).shape == (3,)  # fails because the shape is (3,1)
```

**Comment by vsyrgkanis** [comment]
Yeap, `RegressionForest` does not adhere to the CATE api. It's not a CATE estimator.

**Comment by kbattocchi** [comment]
Despite not being a CATE estimator, it still seems like predict should spit back out something compatible with what was passed to fit.  This is true of nearly all sklearn estimtaors, for example.

**Comment by vsyrgkanis** [comment]
yeah I was lazy not adhering as this was supposed to be an internal used class. I'll fix. This is the case for all GRF, I have added in the docs, that this returns always n_y, irrespective if the input y was a vector, it will always return n_y. So at least the code is consistent with its docs. 

But I agree this is non-standard., e.g. unliek sklearn estimators, so will try to change at least in the regression forest. I wanted to avoid doing all the sub-casing in the causal forest that also always returns n_t and n_y, but maybe I can change there too. 

Will be for next release.
